hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapPutMessageTask.java

```diff
@@ -36,17 +38,17 @@ protected AbstractMapPutMessageTask(ClientMessage clientMessage, Node node, Conn
 
     @Override
     protected void beforeProcess() {
-        startTime = System.currentTimeMillis();
+        startTime = System.nanoTime();
     }
 
     @Override
     protected void beforeResponse() {
-        final long latency = System.currentTimeMillis() - startTime;
+        final long latency = System.nanoTime() - startTime;
         final MapService mapService = getService(MapService.SERVICE_NAME);
         MapContainer mapContainer = mapService.getMapServiceContext().getMapContainer(getDistributedObjectName());
         if (mapContainer.getMapConfig().isStatisticsEnabled()) {
             mapService.getMapServiceContext().getLocalMapStatsProvider().getLocalMapStatsImpl(getDistributedObjectName())
-                    .incrementPuts(latency);
+                    .incrementPuts(NANOSECONDS.toMicros(latency));
```

Comment:
That's what we have our IDEs for :) There is a real-time preview including the parameter name, at least in IDEA (example):
![idea-to-the-rescue](https://user-images.githubusercontent.com/4196298/28278358-1c565256-6b1d-11e7-91ea-615a3ff2c40d.png)